File: 011-020/015.rb
# 最短経路問題
# n×mのグリッドのとき、(n+m)!/n!m!

# 階乗を求める関数
def factrial(num)
  num.zero? ? 1 : factrial(num - 1) * num
end

n = 20
m = 20
puts factrial(n + m) / (factrial(n) * factrial(m))

File: 011-020/015.js
// 最短経路
// n×m => (n + m)! / n!m!

const factrial = num => {
  return num === 0 ? 1 : factrial(num - 1) * num;
};

const n = 20;
const m = 20;

console.log(factrial(n+m)/(factrial(n)*factrial(m)));
